presto-geospatial/src/main/java/com/facebook/presto/geospatial/GeoFunctions.java

```diff
@@ -0,0 +1,493 @@
+    @ScalarFunction("ST_NumPoints")
+    @SqlType(StandardTypes.BIGINT)
+    public static long stNumPoints(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.getEsriGeometry().isEmpty()) {
+            return 0;
+        }
+        else if (geometry.geometryType().equals(POINT)) {
+            return 1;
+        }
+        return ((MultiVertexGeometry) geometry.getEsriGeometry()).getPointCount();
+    }
+
+    @SqlNullable
+    @Description("Returns TRUE if and only if the line is closed and simple")
+    @ScalarFunction("ST_IsRing")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean stIsRing(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_IsRing", LINE_STRING);
+        OGCLineString line = (OGCLineString) geometry;
+        return line.isClosed() && line.isSimple();
+    }
+
+    @Description("Returns the first point of a LINESTRING geometry as a Point")
+    @ScalarFunction("ST_StartPoint")
+    @SqlType(NAME)
+    public static Slice stStartPoint(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_StartPoint", LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        return serialize(OGCGeometry.createFromEsriGeometry(lines.getPoint(0), reference));
+    }
+
+    @Description("Returns the last point of a LINESTRING geometry as a Point")
+    @ScalarFunction("ST_EndPoint")
+    @SqlType(NAME)
+    public static Slice stEndPoint(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_EndPoint", LINE_STRING);
+        MultiPath lines = (MultiPath) geometry.getEsriGeometry();
+        SpatialReference reference = geometry.getEsriSpatialReference();
+        return serialize(OGCGeometry.createFromEsriGeometry(lines.getPoint(lines.getPointCount() - 1), reference));
+    }
+
+    @Description("Return the X coordinate of the point")
+    @ScalarFunction("ST_X")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stX(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_X", POINT);
+        return ((OGCPoint) geometry).X();
+    }
+
+    @Description("Return the Y coordinate of the point")
+    @ScalarFunction("ST_Y")
+    @SqlType(StandardTypes.DOUBLE)
+    public static double stY(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = getGeometry(deserialize(input), "ST_Y", POINT);
+        return ((OGCPoint) geometry).Y();
+    }
+
+    @Description("Returns the closure of the combinatorial boundary of this Geometry")
+    @ScalarFunction("ST_Boundary")
+    @SqlType(NAME)
+    public static Slice stBoundary(@SqlType(NAME) Slice input)
+    {
+        OGCGeometry geometry = deserialize(input);
+        OGCGeometry bound = geometry.boundary();
```

Comment:
nit: this variable seems not needed; if you like to keep it, consider renaming to 'boundary'